pkgs/applications/blockchains/nimbus/default.nix

```
@@ -0,0 +1,64 @@
+{
+  stdenv, fetchgit, lib, nim, which, writeScriptBin,
+  # Options: nimbus_light_client, nimbus_validator_client, nimbus_signing_node
+  makeTargets ? [ "nimbus_beacon_node" ],
+  # WARNING: CPU optmizations that make binary not portable.
+  nativeBuild ? false
```

```suggestion
{ stdenv
, fetchgit
, lib
, nim # version 1.6.10 is known to be stable and overriden in top-level
, which
, writeScriptBin
, makeTargets ? [ "nimbus_beacon_node" ],
  # WARNING: CPU optmizations that make binary not portable.
, nativeBuild ? false
```

makeTargets should be asserted instead to make sure it is not set to anything else.

We generally remove march native without options and gate avx, avx2, etc behind flags. Is that something the compile script support?

---

@SuperSandro2000 what do you mean by "gate"?
@zah can you please comment on options available to use for native compilation?

---

We already set `nativeBuild` to `false` by default because we want to keep the build portable, but if someone wants to for the sake of local optimization they can rebuild with native optimizations enabled. I think that's a nice option to have.

---

This derivation here is already doing a reasonable thing by disabling `-march=native` by default. Our build scripts currently don't offer flags for enabling specific CPU features such as AVX, AVX2, etc. We can add some of these in the future. For now, the only option is to enable `-march=native`.

---

We should turn -march=native off by default but add something like
```
lib.optionalString (stdenv.hostPlatform ? gcc.arch) ''
  -march=${stdenv.hostPlatform.gcc.arch}
''
```
to enable -march when nix is told to optimize software for a specific platform.